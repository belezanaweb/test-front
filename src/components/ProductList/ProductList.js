import React, { useContext, useMemo } from 'react'
import { CartContext } from '../../components/Context/Context'
import Card from '../../components/Card/Card'
import HeaderText from '../../components/HeaderText/HeaderText'
import { currencyFormatter } from '../../utils'
import CartItem from '../../components/CartItem/CartItem'

const ProductList = (props) => {
  const cartContext = useContext(CartContext)
  const { listType } = props
  const renderProductList = useMemo(() => {
    if (Object.keys(cartContext.cartProductsData).length) {
      return cartContext.cartProductsData.items.map((item, index) => {
        return (
          <CartItem
            name={item.product.name}
            imageSrc={item.product.imageObjects[0].small}
            price={
              listType === 'cartScreen'
                ? currencyFormatter.format(item.product.priceSpecification.price)
                : ''
            }
            key={`${item.product.name}${index}`}
          />
        )
      })
    } else {
      return (
        <div>
          <p>O carrinho est√° vazio.</p>
        </div>
      )
    }
  }, [cartContext, listType])

  return (
    <>
      <HeaderText>Produtos</HeaderText>
      <Card>{renderProductList}</Card>
    </>
  )
}

export default ProductList
