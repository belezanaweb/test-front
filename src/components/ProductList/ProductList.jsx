import React from 'react'

export default function ProductList() {
  return (
    <ul>
      <li>3</li>
      <li>2</li>
      <li>1</li>
    </ul>
  )
}
