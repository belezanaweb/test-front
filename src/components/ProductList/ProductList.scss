@import '../../assets/sass/colors';

.product-list {
  width: 100%;
  height: auto;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  margin: 0;
  position: relative;
  border-radius: 3px;
  border: 1px solid #EEEEEE;
  padding: 11px 10px 14px 10px;
  &__thumb {
    width: 65px;
    height: 65px;
    display: block;
    flex: none;
    img {
      width: 100%;
    }
  }
  &__content {
    width: 100%;
    height: auto;
    position: relative;
    margin: 0;
    padding-left: 11px;
  }
  &__title {
    font-size: 13px;
    line-height: 16px;
    font-weight: 400;
    color: $graphiteColor;
  }
  &__price {
    font-weight: 700;
    color: $graphiteColor;
    width: 100%;
    text-align: right;
    display: block;
    margin-top: 15px;
    font-size: 14px;
    line-height: 17px;
  }
}
