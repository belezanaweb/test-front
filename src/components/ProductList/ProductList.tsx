import styled, { css } from 'styled-components';

import { Image, ImageWrapper, ItemPrice } from './ProductListItem';

interface ProductListProps {
  size?: 'sm' | 'md';
}

const ProductList = styled.div<ProductListProps>`


  ${({ size }) => {
    switch (size) {
      case 'sm':
        return css`
          ${ImageWrapper} {
            flex: 0 42px;
          }

          ${Image} {
            height: 42px;
          }

          ${ItemPrice} {
            display: none;
          }
        `;
      case 'md':
      default:
        return css``;
    }
  }}
`;

ProductList.defaultProps = {
  size: 'md',
};

export default ProductList;