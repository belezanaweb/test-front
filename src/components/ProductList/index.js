import React from 'react';
import './styles.css';
import ItemCard from '../ItemCard';
import Loading from '../Loading';

const ProductList = props => {
    return (
        <>
            {props.store.items ? props.store.items.map((element, index) =>
                <ItemCard key={index} item={element} />
            )
                :
                <Loading />
            }
        </>
    )
}

export default ProductList;

