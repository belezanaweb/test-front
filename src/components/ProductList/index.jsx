import React from 'react'

const ProductList = () => {
  return <></>
}

export default ProductList
