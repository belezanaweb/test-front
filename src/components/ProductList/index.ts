export { default } from './ProductList';
export { default as ProductListItem } from './ProductListItem';
