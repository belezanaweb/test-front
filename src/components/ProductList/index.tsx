import React from 'react'
import { useSelector } from 'react-redux'
import { IOrder } from '../../model/order'
import { ProductItemStyled } from '../../pages/ProductBag/styles'
import { RootState } from '../../redux'
import { ContentStyled } from '../../styles/global'

const ProductList: React.FC = () => {
  const order = useSelector<RootState, IOrder | undefined>((state) => {
    return state.orderReducer.order
  })
  return (
    <>
      <h3>Produtos</h3>
      <ContentStyled>
        {order &&
          order.items.length > 0 &&
          order.items.map((item) => {
            return (
              <ProductItemStyled key={item.product.sku}>
                <img src={item.product.imageObjects[0].small} alt={item.product.name} />
                <div>{item.product.name.substring(0, 60)}</div>
              </ProductItemStyled>
            )
          })}
      </ContentStyled>
    </>
  )
}

export default ProductList
