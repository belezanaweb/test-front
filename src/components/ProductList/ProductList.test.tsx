import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import 'jest-styled-components';

import { provider } from '../../utils/testUtils';

import cart from '../../store/models/__mocks__/cart.json';

import ProductList from './ProductList';
import ProductListItem from './ProductListItem';

afterEach(function() {
  cleanup();
});

function renderProducts() {
  return cart?.items?.map(({ product }) => (
    <ProductListItem key={product.sku} data={product} />
  ));
}

describe('<ProductList />', () => {
  it('should not render when data is empty.', () => {
    const { container } = provider(<ProductList />, render);

    expect(container.firstChild).toBeEmpty();
  });

  it('should render within two products.', () => {
    const { container } = provider((
      <ProductList>
        {renderProducts()}
      </ProductList>
    ), render);

    expect(container).toMatchSnapshot();
  });

  it('should render ProductListItem on "sm" size.', () => {
    const { container } = provider((
      <ProductList size="sm">
        {renderProducts()}
      </ProductList>
    ), render);

    expect(container).toMatchSnapshot();
  });
});
