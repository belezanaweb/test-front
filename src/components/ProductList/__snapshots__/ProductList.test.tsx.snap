// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> should render ProductListItem on "sm" size. 1`] = `
.c1 {
  background-color: transparent;
  border: 1px solid #CCC;
  border-radius: 3px;
  padding: 15px 13px;
  border-color: #EEE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 15px;
  padding: 11px 11px 12px;
}

.c1:last-of-type {
  margin-bottom: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 65px;
  -ms-flex: 0 65px;
  flex: 0 65px;
}

.c5 {
  height: 65px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 0 0 10px;
}

.c7 {
  color: #999;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Helvetica Neue",sans-serif;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: -0.015em;
  -moz-letter-spacing: -0.015em;
  -ms-letter-spacing: -0.015em;
  letter-spacing: -0.015em;
  color: #212122;
  font-size: 13px;
}

.c9 {
  color: #999;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Helvetica Neue",sans-serif;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: -0.015em;
  -moz-letter-spacing: -0.015em;
  -ms-letter-spacing: -0.015em;
  letter-spacing: -0.015em;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  color: #212122;
  font-weight: 700;
  -webkit-letter-spacing: -0.075em;
  -moz-letter-spacing: -0.075em;
  -ms-letter-spacing: -0.075em;
  letter-spacing: -0.075em;
  line-height: 17px;
}

.c0 .c2 {
  -webkit-flex: 0 42px;
  -ms-flex: 0 42px;
  flex: 0 42px;
}

.c0 .c4 {
  height: 42px;
}

.c0 .c8 {
  display: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="sc-AxjAm c1"
    >
      <div
        class="c2 c3"
      >
        <img
          alt="Imagem Ilustrativa"
          class="c4 c5"
          src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png"
        />
      </div>
      <div
        class="c6"
      >
        <span
          class="sc-AxirZ c7"
        >
          L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g
        </span>
        <span
          aria-label="price"
          class="sc-AxirZ c8 c9"
        >
          R$ 225,90
        </span>
      </div>
    </div>
    <div
      class="sc-AxjAm c1"
    >
      <div
        class="c2 c3"
      >
        <img
          alt="Imagem Ilustrativa"
          class="c4 c5"
          src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/good-girl-carolina-herrera-eau-de-parfum-perfume-feminino-30ml-38273-1960525940762131267.jpg"
        />
      </div>
      <div
        class="c6"
      >
        <span
          class="sc-AxirZ c7"
        >
          Good Girl Carolina Herrera Eau de Parfum - Perfume Feminino 30ml
        </span>
        <span
          aria-label="price"
          class="sc-AxirZ c8 c9"
        >
          R$ 299,00
        </span>
      </div>
    </div>
    <div
      class="sc-AxjAm c1"
    >
      <div
        class="c2 c3"
      >
        <img
          alt="Imagem Ilustrativa"
          class="c4 c5"
          src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/senscience-inner-restore-intesif-deep-repairing-masque-mascara-de-tratamento-50ml-3019-7903251722539384904.png"
        />
      </div>
      <div
        class="c6"
      >
        <span
          class="sc-AxirZ c7"
        >
          Senscience Inner Restore Intensif - Máscara Capilar 50ml
        </span>
        <span
          aria-label="price"
          class="sc-AxirZ c8 c9"
        >
          R$ 99,90
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductList /> should render within two products. 1`] = `
.c0 {
  background-color: transparent;
  border: 1px solid #CCC;
  border-radius: 3px;
  padding: 15px 13px;
  border-color: #EEE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 15px;
  padding: 11px 11px 12px;
}

.c0:last-of-type {
  margin-bottom: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 65px;
  -ms-flex: 0 65px;
  flex: 0 65px;
}

.c2 {
  height: 65px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 0 0 10px;
}

.c4 {
  color: #999;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Helvetica Neue",sans-serif;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: -0.015em;
  -moz-letter-spacing: -0.015em;
  -ms-letter-spacing: -0.015em;
  letter-spacing: -0.015em;
  color: #212122;
  font-size: 13px;
}

.c5 {
  color: #999;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Helvetica Neue",sans-serif;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: -0.015em;
  -moz-letter-spacing: -0.015em;
  -ms-letter-spacing: -0.015em;
  letter-spacing: -0.015em;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  color: #212122;
  font-weight: 700;
  -webkit-letter-spacing: -0.075em;
  -moz-letter-spacing: -0.075em;
  -ms-letter-spacing: -0.075em;
  letter-spacing: -0.075em;
  line-height: 17px;
}

<div>
  <div
    class=""
  >
    <div
      class="sc-AxjAm c0"
    >
      <div
        class="c1"
      >
        <img
          alt="Imagem Ilustrativa"
          class="c2"
          src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png"
        />
      </div>
      <div
        class="c3"
      >
        <span
          class="sc-AxirZ c4"
        >
          L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g
        </span>
        <span
          aria-label="price"
          class="sc-AxirZ c5"
        >
          R$ 225,90
        </span>
      </div>
    </div>
    <div
      class="sc-AxjAm c0"
    >
      <div
        class="c1"
      >
        <img
          alt="Imagem Ilustrativa"
          class="c2"
          src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/good-girl-carolina-herrera-eau-de-parfum-perfume-feminino-30ml-38273-1960525940762131267.jpg"
        />
      </div>
      <div
        class="c3"
      >
        <span
          class="sc-AxirZ c4"
        >
          Good Girl Carolina Herrera Eau de Parfum - Perfume Feminino 30ml
        </span>
        <span
          aria-label="price"
          class="sc-AxirZ c5"
        >
          R$ 299,00
        </span>
      </div>
    </div>
    <div
      class="sc-AxjAm c0"
    >
      <div
        class="c1"
      >
        <img
          alt="Imagem Ilustrativa"
          class="c2"
          src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/4/senscience-inner-restore-intesif-deep-repairing-masque-mascara-de-tratamento-50ml-3019-7903251722539384904.png"
        />
      </div>
      <div
        class="c3"
      >
        <span
          class="sc-AxirZ c4"
        >
          Senscience Inner Restore Intensif - Máscara Capilar 50ml
        </span>
        <span
          aria-label="price"
          class="sc-AxirZ c5"
        >
          R$ 99,90
        </span>
      </div>
    </div>
  </div>
</div>
`;
